### PR TITLE
Star dylan patch 1

### PR DESCRIPTION
**Pull request by StarDylan**
Cleaned up the code a bit and added temp and status.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,54 @@
 # pyPhin
```
Comment (starcode911) [inline README.md:2]: Add you copyright information and the github link to this projects

File: pyPhin.py [hunk 2]
```diff
@@ -1,19 +1,26 @@
+#!/usr/bin/env python3
+"""
+pHin Unoffical Python API
```
Comment (starcode911) [inline pyPhin.py]: As some comments about pHin (eg, pHin Smart Water Monitor and point to the website)

File: pyPhin.py [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
+Copyright (C) 2020 StarDylan
```
Comment (starcode911) [inline pyPhin.py:4]: Add the github url to this repository

File: pyPhin.py [hunk 4]
```diff
@@ -109,47 +114,76 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
...
+	def createHeader(self, deviceUUID, authToken=None, version="1.0.0"):
+		headers = {"x-phin-concise":"true",
+			"x-phin-reporting-app-id":"ios-app",
+			"x-phin-reporting-device-id":deviceUUID}
+		headers["Accept-Version"] = version
+		if authToken != None:
+			headers["Authorization"] = "Bearer " + authToken
+		return headers
+
+	def checkRequest(self, json):
```
Comment (starcode911) [inline pyPhin.py]: what if "json" is not actual json.. but None or a int? first check if it's actual json

File: pyPhin.py [hunk 5]
```diff
@@ -109,47 +114,76 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
...
+		if not json["success"]:
+			raise Exception(json)
+	def checkUrlRoute(self, urlRoute):
+		if re.match("^\/",urlRoute) == None:
```
Comment (starcode911) [inline pyPhin.py:221]: first check if urlRoute.. might be None or an array or ...

File: pyPhin.py [hunk 6]
```diff
@@ -109,47 +114,76 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
...
+			"status":"balanced"
+		}
+		'''
+		return data
 
+	def createHeader(self, deviceUUID, authToken=None, version="1.0.0"):
+		headers = {"x-phin-concise":"true",
+			"x-phin-reporting-app-id":"ios-app",
+			"x-phin-reporting-device-id":deviceUUID}
+		headers["Accept-Version"] = version
```
Comment (starcode911) [inline pyPhin.py]: why not include this header as part of the dictionary creation above?

File: pyPhin.py [hunk 7]
```diff
@@ -32,34 +43,32 @@ def login(self, contact, deviceUUID):
...
 
-	def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
-
-		baseHeaders = {"x-phin-concise":"true",
-			"x-phin-reporting-app-id":"ios-app",
-			"x-phin-reporting-device-id":deviceUUID}
+	def verify(self, contact, deviceUUID, verifyUrl, verificationCode):
 
+		if not verificationCode.isnumeric():
+			raise Exception("Verification Code is not Numeric!")
```
Comment (starcode911) [inline pyPhin.py]: move into a function also check for valid type (eg, not None)

File: pyPhin.py [hunk 8]
```diff
@@ -32,34 +43,32 @@ def login(self, contact, deviceUUID):
...
+		self.checkUrlRoute(verifyUrl)
+		self.checkEmail(contact)
 
 		''' verify_route
-		{
+		{n
```
Comment (starcode911) [inline pyPhin.py]: n?

File: pyPhin.py [hunk 9]
```diff
@@ -109,47 +118,80 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
 		'''
 		req = requests.get(
 			self.baseUrl+locationUrl,
-			headers=self.createHeader(authToken, deviceUUID, "2.0.1")
+			headers=self.createHeader(deviceUUID, authToken, "2.0.1")
```
Comment (starcode911) [inline pyPhin.py:129]: what is this version used for? Maybe put in a constant?

File: pyPhin.py [hunk 10]
```diff
@@ -109,47 +118,80 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
...
 			)
 		reqJson = json.loads(req.text)
 
-		if not reqJson["success"]:
-			raise Exception(reqJson)
+		self.checkRequest(req.text)
```
Comment (starcode911) [inline pyPhin.py]: Should this not be *before* json.load? also, what if req is None?

File: pyPhin.py [hunk 11]
```diff
@@ -109,47 +118,80 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
...
 
-		return reqJson["vessels"][0]["waterReport"][waterQualityType]["value"]
 
+		self.checkRequest(req.text)
+		reqJson = json.loads(req.text)
 
+		data = {}
 
+		for dataType in ["TA","CYA","TH"]:
+			data[dataType] = reqJson["vessels"][0]["waterReport"][dataType]["value"]
```
Comment (starcode911) [inline pyPhin.py]: What if pHin decide not return value, eg no "CYA" comes back... in that case

data[dataType] = reqJson["vessels"][0]["waterReport"][dataType] == None

and thus your code errors out..

**Review by starcode911 — COMMENTED** [review]
looking good!

File: pyPhin.py (hunk 9 above)
Comment (StarDylan) [reply pyPhin.py:129]: Each page has a different version associated with it.

File: pyPhin.py (hunk 11 above)
Comment (StarDylan) [reply pyPhin.py:162]: Well thats a bit extreme. It would say `KeyError: CYA` , and instead of me typing an exception saying `Exception: Cannot Find CYA`.

**Review by starcode911 — APPROVED** [review]
looks good

**Pull request merged**